localstack/services/apigateway/exporter.py

```diff
@@ -0,0 +1,285 @@
+                    response = {"description": f"{status_code} response"}
+                    content = {}
+                    if response_parameters := values.get("responseParameters"):
+                        headers = {}
+                        for parameter in response_parameters:
+                            in_, name = parameter.removeprefix("method.response.").split(".")
+                            # TODO: other type? query?
+                            if in_ == "header":
+                                headers[name] = {"schema": {"type": "string"}}
+
+                        if headers:
+                            response["headers"] = headers
+                    if response_models := values.get("responseModels"):
+                        for content_type, model_name in response_models.items():
+                            content[content_type] = {
+                                "schema": {"$ref": f"#/components/schemas/{model_name}"}
+                            }
+                    if integration_response := integration_responses.get(status_code, {}):
+                        produces.update(integration_response.get("responseTemplates", {}).keys())
+
+                    response["content"] = content
+                    responses[status_code] = response
+
+                request_parameters = method_config.get("requestParameters", {})
+                parameters = []
+                for parameter, required in request_parameters.items():
+                    in_, name = parameter.removeprefix("method.request.").split(".")
+                    in_ = in_ if in_ != "querystring" else "query"
+                    parameters.append({"name": name, "in": in_, "schema": {"type": "string"}})
+
+                request_body = {"content": {}}
+                request_models = method_config.get("requestModels", {})
+                for content_type, model_name in request_models.items():
+                    request_body["content"][content_type] = {
+                        "schema": {"$ref": f"#/components/schemas/{model_name}"},
+                    }
+                    request_body["required"] = True
+
+                method_operations = {"responses": responses}
+                if parameters:
+                    method_operations["parameters"] = parameters
+                if request_body["content"]:
+                    method_operations["requestBody"] = request_body
+                if operation_name := method_config.get("operationName"):
+                    method_operations["operationId"] = operation_name
+                if with_extension and method_integration:
+                    method_operations["x-amazon-apigateway-integration"] = self._get_integration(
+                        method_integration
+                    )
+
+                spec.path(path=path, operations={method: method_operations})
+
+    def export(self, api_id: str, stage: str, export_format: str, with_extension: bool) -> str:
```

Comment:
nit: Seems like the return type would be `str | dict`, depending on which `export_format` is specified

Reply:
Yes, definitely. I guess we should move the logic of the JSON dumps inside the exporter instead of the provider, this could be done in a follow up. Updated the signature in the meantime, thanks a lot for the thorough review.